x-pack/plugins/fleet/server/services/agents/uploads.ts

```diff
@@ -97,10 +86,48 @@ async function _getRequestDiagnosticsActions(
     },
   });
 
-  return res.hits.hits.map((hit) => ({
-    actionId: hit._source?.action_id as string,
-    timestamp: hit._source?.['@timestamp'],
-  }));
+  const agentActionIds = agentActionRes.hits.hits.map((hit) => hit._source?.action_id as string);
+
+  if (agentActionIds.length === 0) {
+    return [];
+  }
+
+  try {
+    const actionResults = await esClient.search<any>({
+      index: AGENT_ACTIONS_RESULTS_INDEX,
+      ignore_unavailable: true,
+      size: SO_SEARCH_LIMIT,
+      query: {
+        bool: {
+          must: [
+            {
+              terms: {
+                action_id: agentActionIds,
+              },
+            },
+            {
+              term: {
+                agent_id: agentId,
+              },
+            },
+          ],
+        },
+      },
+    });
+    return actionResults.hits.hits.map((hit) => ({
+      actionId: hit._source?.action_id as string,
+      timestamp: hit._source?.['@timestamp'],
+      fileId: hit._source?.data?.file_id as string,
```

Comment:
taking the `file_id` from action results, this part is not yet implemented, asked here: https://github.com/elastic/elastic-agent/pull/1631#issuecomment-1298621934